website_anchor_smooth_scroll/static/src/js/website_anchor_smooth_scroll.js

```diff
@@ -0,0 +1,27 @@
+/**
+*    Copyright 2016 Antiun Ingeniería S.L. - Jairo Llopis
+*    Copyright 2016 LasLabs Inc.
+*    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+**/
+
+odoo.define('website_anchor_smooth_scroll.enable', function(require) {
+    'use strict';
+
+    var base = require('web_editor.base');
+
+    var smooth_scroll = function(event) {
+        event.preventDefault();
+        var anchor_fragment = event.target.hash;
+
+        // Do this before scrolling so that browser history accurately reflects scroll position at time of click 
+        history.pushState(null, document.title, anchor_fragment);
+
+        $('html, body').stop().animate({
```

Comment:
In v8, outside code could only get access to and make meaningful use of the promise by calling `website_anchor_smooth_scroll()` directly, so the equivalent here would involve exporting `smooth_scroll()` from the module. Is that what you have in mind?